### PR TITLE
rules 1011 2008 (split from original)

### PR DESCRIPTION
**Pull request by harleenkohli**
NOTE: Possible issue on how to write tests for one of the rules (See comments)

In this PR:

- [In Progress]	House keeping changes for all rules:
	- change ruleID (if needed)
	- change ruleID in filenames (if needed)
	- change rulename (if needed)
	- change rulename in filenames (if needed)
	- rename resx strings (description)
	- rename resx strings (messages)
	- rename/regenerate test cases if needed.

status 		|	old	 	| new
|---|---|---|
[done ] 	|	sarif1001	| sarif1001
[done ] 	|	sarif1003	| sarif1002
[done ] 	|	sarif1007	| sarif1006
[done ] 	|	sarif1008	| sarif2001
[ ]		|	sarif1012	| sarif1007 (combine 1)
[ ] 		|	sarif1013	| sarif1007 (combine 1)
[ ] 		|	sarif1014	| sarif1004 (combine 2)
[done ] 	|	sarif1015	| sarif1005
[done ] 	|	sarif1016	| sarif1008
[done ] 	|	sarif1017	| sarif1009
[ ] 		|	sarif1018	| sarif1004 (combine 2)
[done ] 	|	sarif1019	| sarif1010
[thisPR]	|	sarif1020	| sarif1011 (split)
[thisPR]	|	sarif1020	| sarif2008 (split)

### CODE REVIEW

File: src/Sarif.Multitool/ValidateCommand.cs [hunk 1]
```diff
@@ -60,6 +60,8 @@ protected override SarifValidationContext CreateContext(ValidateOptions options,
 
             if (!string.IsNullOrEmpty(sarifText))
             {
+                // DISCUSS IN PR: Deserializing the object here injects "$Schema" property, corrupting our test for
+                // SARIF2008_ProvideSchema
```
Comment (harleenkohli) [inline src/Sarif.Multitool/ValidateCommand.cs:64]: Larry - need to discuss how to solve this.

Comment (harleenkohli) [reply src/Sarif.Multitool/ValidateCommand.cs:64]: This is also a possible bug in our Validate command. Isnt it? If you confirm, i can log a bug, mark the test cases as ignore and move on for now.

---
In reply to: [443040878](https://github.com/microsoft/sarif-sdk/pull/1921#discussion_r443040878) [](ancestors = 443040878)

**Comment by lgolding** [comment]
> using System;

System includes first.

---
Refers to: src/Sarif.Multitool/Rules/SARIF1011.ReferenceFinalSchema.cs:5 in 65c0b4d. [](commit_id = 65c0b4db4c37669eb71975d17fb110afb1772d5a, deletion_comment = False)

File: src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class ProvideSchema : SarifValidationSkimmerBase
+    {
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF2008_ProvideSchema_FullDescription_Text
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Error;
```
Comment (lgolding) [inline src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs:17]: >Error [](start = 66, length = 5)

Warning (so rename resource strings accordingly).

Comment (lgolding) [reply src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs:17]: `run.$schema` is optional, so not error-level.

---
In reply to: [443047241](https://github.com/microsoft/sarif-sdk/pull/1921#discussion_r443047241) [](ancestors = 443047241)

Comment (lgolding) [reply src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs:17]: Oh, I see that the resources _do_ say "warning".

---
In reply to: [443047241](https://github.com/microsoft/sarif-sdk/pull/1921#discussion_r443047241) [](ancestors = 443047241)

Comment (harleenkohli) [reply src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs:17]: Nice catch!!!

---
In reply to: [443047522](https://github.com/microsoft/sarif-sdk/pull/1921#discussion_r443047522) [](ancestors = 443047522,443047241)

File: src/Sarif.Multitool/ValidateCommand.cs (hunk 1 above)
Comment (lgolding) [reply src/Sarif.Multitool/ValidateCommand.cs]: Yes, change to SARIF2008 is the right fix.

---
In reply to: [443046418](https://github.com/microsoft/sarif-sdk/pull/1921#discussion_r443046418) [](ancestors = 443046418,443040878)

File: src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        public override string Id => RuleId.ProvideSchema;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF2008_ProvideSchema_Warning_Default_Text)
+        };
+
+        protected override void Analyze(SarifLog log, string logPointer)
+        {
+            if (!Context.InputLogToken.HasProperty("$schema"))
```
Comment (lgolding) [inline src/Sarif.Multitool/Rules/SARIF2008.ProvideSchema.cs:26]: >) [](start = 61, length = 1)

Yes!

**Pull request merged**